components/post_view/message_attachments/action_button/action_button.jsx

```diff
@@ -3,25 +3,36 @@
 
 import PropTypes from 'prop-types';
 import React from 'react';
+import {changeOpacity} from 'mattermost-redux/utils/theme_utils';
 
 import Markdown from 'components/markdown';
+import {STATUS_COLORS} from 'utils/constants';
 
 export default class ActionButton extends React.PureComponent {
     static propTypes = {
         action: PropTypes.object.isRequired,
         handleAction: PropTypes.func.isRequired,
         disabled: PropTypes.bool,
+        theme: PropTypes.object.isRequired,
     }
 
     render() {
-        const {action, handleAction, disabled} = this.props;
+        const {action, handleAction, disabled, theme} = this.props;
+        let customButtonStyle;
+
+        if (action.style) {
+            const hexColor = STATUS_COLORS[action.style] || theme[action.style] || action.style;
+            customButtonStyle = {borderColor: changeOpacity(hexColor, 0.25), backgroundColor: '#ffffff', color: hexColor, borderWidth: 2};
```

Comment:
Wonder what happens if hexColor is undefined or a non valid color.

Reply:
Like in the mobile PR, I assumed that the server will send me a valid style. I can make some validation here, but then we should do the same in the mobile app.

Reply:
If a "wrong color" just shows the button as black or default, I am OK with that. My concern is whether it would crash the application, or things like that.
If you feel it would make things crash, it would be great if you could add the validation both here and in mobile.

Reply:
@larkox, I added extra validation to check if the provided color is correct. Hope it's ok :)